### PR TITLE
Customer Cloud Deployment documentation

### PR DESCRIPTION
**Pull request by bvdeenen**
First version, only with all infrastructure enabled in the Helm chart

### CODE REVIEW

File: docs/quickstart/ccd.md [hunk 1]
```diff
@@ -0,0 +1,246 @@
+---
+title: Customer Cloud Deployments
+hide_table_of_contents: false
+---
+
+[github]: https://github.com/strmprivacy/data-plane-helm-chart/tree/master
+[oma]: https://backstage.strm.in/docs/default/component/organization-management-and-access/
+[jk]: https://docs.google.com/document/d/1ymb8JPaOflX8GZzICV6WCE_L0GGLRrBAPtk0dgeHJdc/edit
```
Comment (ivan-p92) [inline docs/quickstart/ccd.md:8]: These two can be removed

File: docs/quickstart/ccd.md [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+[profile]: https://console.strmprivacy.io/upgrading
+[values]: https://console.strmprivacy.io/installation/configuration
+[confluent]: https://docs.confluent.io/platform/current/quickstart/ce-docker-quickstart.html#cp-quickstart-step-1
+[makefile]: https://github.com/strmprivacy/data-plane-helm-chart/blob/master/Makefile
+
+This hands-on sessions shows how to get up-and-running with your Customer Cloud Deployment, and verify its
+functionality.
+
+## Get an account with a 'Self Hosted' subscription
+Your [Strm profile page][profile] should show: "Current subscription: Self hosted". Get in touch with your Strm
```
Comment (ivan-p92) [inline docs/quickstart/ccd.md:21]: `Strm` should be all-caps here and elswehere in this quickstart.

File: docs/quickstart/ccd.md [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+representative if you're on a Free or Business subscription.
+
+Once you have this profile active, you can start.
```
Comment (ivan-p92) [inline docs/quickstart/ccd.md:24]:
```suggestion
Once you're on a self hosted subscription, you can proceed with this quickstart guide.
```

File: docs/quickstart/ccd.md [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+* [`telepresence`][telepresence]. This tool allows your development computer to become _part of_ the k8s cluster, so
+  that you can directly access k8s services.
+* [`k9s`](https://github.com/derailed/k9s) (optional). This _textual user interface_ offers a very convenient way to interact with
+  kubernetes clusters.
+* [`kubectx and kubens`](https://github.com/ahmetb/kubectx) (optional). Very useful tools to switch the default
+  kubernetes context and namespace.
+
+## Start the installation of the Strm customer data-plane
+
+1. download the credentials file `values.yaml` [here][values] into the `data-plane-helm-chart` directory.
```
Comment (ivan-p92) [inline docs/quickstart/ccd.md:42]: I think this should be the last step in this list.

File: docs/quickstart/ccd.md [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+1. make sure you have active kubernetes credentials to a cluster: `kubectl get nodes` should show the nodes of your
+   cluster.
+1. Create a namespace `strmprivacy` and set that as default (`kubens strmprivacy`)
```
Comment (ivan-p92) [inline docs/quickstart/ccd.md:45]: Perhaps we can add the kubectl command to create the namespace.
```suggestion
1. Create a namespace `strmprivacy` (`kubectl create namespace strmprivacy`) and set that as default (`kubens strmprivacy`)
```

File: docs/quickstart/ccd.md [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+      installationClientSecret: "tii...."
+
+    kafka:
+      enabled: true
+
+    redis:
+      enabled: true
+
+    postgresql:
+      enabled: true
```
Comment (ivan-p92) [inline docs/quickstart/ccd.md:67]: We can make this a yaml listing:
```suggestion
The `values.yaml` file should be similar to this:

```yaml
  registry:
    imagePullSecret: "ewogIC...."

  license:
    installationId: "f4cea...."
    installationClientId: "ins-...."
    installationClientSecret: "tii...."

  kafka:
    enabled: true

  redis:
    enabled: true

  postgresql:
    enabled: true
```
```

File: docs/quickstart/ccd.md [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+
+**this is broken, don't use it**
+
+You'll have to add `namespace=...` to every `make` command (or modify the `Makefile`)
+
+    make namespace namespace=mycustomnamespace
+    make install namespace=mycustomnamespace
+
+The install seems to work correctly but trying to send events seems to go into some black
+hole. *to be investigated*
```
Comment (ivan-p92) [inline docs/quickstart/ccd.md:88]: I think we can phrase this differently:
```suggestion
:::warning
This is still a work in progress, so do not use it yet.
:::

Add `namespace=...` to every `make` command (or modify the `Makefile`)

    make namespace namespace=mycustomnamespace
    make install namespace=mycustomnamespace
```

File: docs/quickstart/ccd.md [hunk 8]
```diff
@@ -0,0 +1,246 @@
...
+    NAME                                             READY   UP-TO-DATE   AVAILABLE   AGE
+    decrypter-a268aea6-4b5b-4241-b833-9a84f4f44bc4   1/1     1            1           93m
+
+You could use `kubectl describe deployment decrypter...` to see the annotations on the deployment, and see that it is
+indeed processing your `test-2` stream
+
+### Sending events with the cli
+
+In order to use the cli to simulated data, you'll need to send the events via telepresence, using the `events-api-url`
+parameter (that you could also store in the strm config file (`strm context config`).
```
Comment (ivan-p92) [inline docs/quickstart/ccd.md:142]:
```suggestion
To simulate events with our cli, the `events-api-url` paramater must be set to the host exposed via telepresence (the url can also be set in the strm config file (`strm context config`).
```

File: docs/quickstart/ccd.md [hunk 9]
```diff
@@ -0,0 +1,246 @@
...
+
+    {"strmMeta": {"eventContractRef": "strmprivacy/example/1.3.0", "nonce": -1742873135, "timestamp": 1652181297380, "keyLink": "556c1be7-4332-4058-9d36-5e3e5a66e121", "billingId": "strmprodccdtest1908747604", "consentLevels": [0, 1, 2]}, "uniqueIdentifier": "unique-81", "consistentValue": "session-922", "someSensitiveValue": "AQM0jlnxbeNZSJzvJWLpnMjyYET1Jb1Yz+5yZVB5i6Dq", "notSensitiveValue": "not-sensitive-78"}
+
+### Kafka
+Install [Confluent client tools][confluent]. You don't have to start Confluent, you only need to add the `bin` directory
+of the unpacked confluent tar file  to your `$PATH`.
+
+
+> There is currently no supported way for non Strmprivacy engineers to determine the Kafka topic name.
+  Exposing this information is a priority.
```
Comment (ivan-p92) [inline docs/quickstart/ccd.md:169]:
```suggestion
:::note
Currently, only STRM Privacy engineers can determine the Kafka topic name.
Exposing this information is a priority. For now, ask a STRM engineer for this information.
:::
```

File: docs/quickstart/ccd.md [hunk 10]
```diff
@@ -0,0 +1,246 @@
...
+
+You can access the input topic of a decrypter as follows
+
+    kubectl get deployments.apps decrypter-... -o=jsonpath='{.metadata.annotations.input-topic}'
+
+    stream-a268aea6-4b5b-4241-b833-9a84f4f44bc4
+
+I haven't thought of a way to access the topic of a derived stream yet. Ask the Strm support engineers, they can access
+the production database to find the topic name.
```
Comment (ivan-p92) [inline docs/quickstart/ccd.md:178]: With above suggestion, I think we can remove this.

File: docs/quickstart/ccd.md [hunk 11]
```diff
@@ -0,0 +1,246 @@
+---
+title: Customer Cloud Deployments
+hide_table_of_contents: false
+---
+
+[github]: https://github.com/strmprivacy/data-plane-helm-chart/tree/master
+[oma]: https://backstage.strm.in/docs/default/component/organization-management-and-access/
+[jk]: https://docs.google.com/document/d/1ymb8JPaOflX8GZzICV6WCE_L0GGLRrBAPtk0dgeHJdc/edit
+[telepresence]: https://www.telepresence.io/
+[sinks]: https://docs.strmprivacy.io/docs/latest/quickstart/receiving-s3/#create-sink
```
Comment (ivan-p92) [inline docs/quickstart/ccd.md:10]: Updated ref for suggestions to S3 section:
```suggestion
[data-connector]: https://docs.strmprivacy.io/docs/latest/quickstart/batch-exporter/#creating-a-data-connector
```

File: docs/concepts/ccd.md [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+[console]: https://console.strmprivacy.io
+[cli]: https://docs.strmprivacy.io/docs/latest/cli-index/
+[api]: https://github.com/strmprivacy/api-definitions/tree/master/protos/strmprivacy
+
+In a Customer Cloud Deployment (CCD), _all event processing takes place in a Kubernetes cluster specified by the
+customer_. **Your sensitive events never leave your own cluster.**
+
+The entities doing the processing get configured by Strmprivacy controls _outside that cluster_, in the Strmprivacy
+cloud. Inside the _customer data plane_ several _agents_ are running that query `api.strmprivacy.io` every few seconds.
+
```
Comment (jankeesvanandel) [inline docs/concepts/ccd.md:14]: Maybe add an extra sentence that this is part of our control plane/management layer and only contains configuration data. Maybe even add a graphic as well to make it absolutely clear that there is a separation of data plane and control plane.

File: docs/quickstart/ccd.md [hunk 13]
```diff
@@ -0,0 +1,246 @@
...
+### Exporting to an s3 bucket
+
+Create the s3 bucket and the associated `sink` (soon to be renamed to `data-connector`). See [docs][sinks] for details
+about the setup.
+
+    strm creat sink s3 --sink-type S3 .... // see [docs][sinks]
+
+    strm create batch-exporter test-2 --sink s3 --path-prefix ccd-prod-ovh
+
+You should see a batch-exporter come to live:
```
Comment (ivan-p92) [inline docs/quickstart/ccd.md:200]: Updated for data connectors:

```suggestion
### Exporting to an S3 bucket

Create the S3 bucket and the associated `data connector`. See [docs][data-connector] for details
about the credentials configuration.

    strm create data-connector s3 s3-connector --credentials-file=...

    strm create batch-exporter test-2 --data-connector s3-connector --path-prefix ccd-prod-ovh

You should see a newly-created batch-exporter deployment:
```

File: docs/concepts/ccd.md [hunk 14]
```diff
@@ -0,0 +1,20 @@
...
+---
+[console]: https://console.strmprivacy.io
+[cli]: https://docs.strmprivacy.io/docs/latest/cli-index/
+[api]: https://github.com/strmprivacy/api-definitions/tree/master/protos/strmprivacy
+
+In a Customer Cloud Deployment (CCD), _all event processing takes place in a Kubernetes cluster specified by the
+customer_. **Your sensitive events never leave your own cluster.**
+
+The entities doing the processing get configured by Strmprivacy controls _outside that cluster_, in the Strmprivacy
+cloud. Inside the _customer data plane_ several _agents_ are running that query `api.strmprivacy.io` every few seconds.
```
Comment (jankeesvanandel) [inline docs/concepts/ccd.md:13]: Maybe rename query to poll?

**Review by ivan-p92 — CHANGES_REQUESTED** [review]
Added a bunch of suggestions (can be applied in batch).

File: docs/quickstart/ccd.md [hunk 15]
```diff
@@ -0,0 +1,246 @@
...
+representative if you're on a Free or Business subscription.
+
+Once you have this profile active, you can start.
+
+## Install the following tools
+
+* `make` (optional). The Helm repository contains handy `make` targets such as _install_, _upgrade_, _wipe_. None of
+  these are mandatory, one can also execute the commands in the [`Makefile`][makefile] by hand.
+* [`kubectl`](https://kubernetes.io/docs/tasks/tools/)
+* [`helm`](http://helm.sh). This Kubernetes package manager is used for installing (and upgrading) your Strm customer
```
Comment (jankeesvanandel) [inline docs/quickstart/ccd.md:31]: Maybe explicitly state that helm is mandatory.

File: docs/quickstart/ccd.md [hunk 16]
```diff
@@ -0,0 +1,246 @@
...
+  dataplane.
+* [`telepresence`][telepresence]. This tool allows your development computer to become _part of_ the k8s cluster, so
```
Comment (jankeesvanandel) [inline docs/quickstart/ccd.md:33]: This is optional, right?

File: docs/quickstart/ccd.md (hunk 4 above)
Comment (jankeesvanandel) [inline docs/quickstart/ccd.md:42]: The link "here" is almost unreadable, partly because it's next to another blue text. I missed the link completely.
Maybe rephrase it to "from the console", to make it stand out a bit more.

File: docs/concepts/ccd.md [hunk 17]
```diff
@@ -0,0 +1,20 @@
...
+
+> ‼ this requires that an outgoing firewall rule allows access to `api.strmprivacy.io` on port 443. All communication is TLS
+encrypted, and defined by a [public gRPC protocol][api]
+
+The agents are indirectly controlled via the [Strmprivacy Console][console] and/or the [cli][cli].
+
```
Comment (jankeesvanandel) [inline docs/concepts/ccd.md:19]: I think it's good to mention that we've tested and it works on multiple types of infra: Local k8s, GKE, EKS, AKS, OVH...

File: docs/quickstart/ccd.md (hunk 7 above)
Comment (jankeesvanandel) [inline docs/quickstart/ccd.md:88]: This text should really be removed. It looks very clumsy to first say something is broken, then explain how it works and then again mention that it doesn't work.
For now just add the text that installing in another namespace than `strmprivacy` is not yet fully operational.

File: docs/quickstart/ccd.md [hunk 18]
```diff
@@ -0,0 +1,246 @@
...
+
+## Trying again
+
+If you've made mistakes and want to start over:
+
+1. `make uninstall` cleanly uninstalls the helm release
+1. `kubectl delete ns strmprivacy` kills everything (including the k8s namespace). Don't forget to recreate the
+   namespace afterwards.
+
+`make wipe` combines the above two steps
```
Comment (jankeesvanandel) [inline docs/quickstart/ccd.md:98]: Maybe this chapter should be placed at the bottom? I'm not sure.

File: docs/quickstart/ccd.md [hunk 19]
```diff
@@ -0,0 +1,246 @@
...
+> There is currently no supported way for non Strmprivacy engineers to determine the Kafka topic name.
+  Exposing this information is a priority.
+
+You can access the input topic of a decrypter as follows
+
+    kubectl get deployments.apps decrypter-... -o=jsonpath='{.metadata.annotations.input-topic}'
+
+    stream-a268aea6-4b5b-4241-b833-9a84f4f44bc4
+
+I haven't thought of a way to access the topic of a derived stream yet. Ask the Strm support engineers, they can access
```
Comment (jankeesvanandel) [inline docs/quickstart/ccd.md:177]: There should not be "I" in such a text as we speak on behalf of the company. Also, there is no visible author, so who is "I"?

File: docs/quickstart/ccd.md (hunk 10 above)
Comment (jankeesvanandel) [inline docs/quickstart/ccd.md:178]: Let's remove the explicit "access the production database to".

File: docs/quickstart/ccd.md [hunk 20]
```diff
@@ -0,0 +1,246 @@
...
+    5afc9c0a-de67-4e38-bd0a-66640f4349c7    {"strmMeta":{"eventContractRef":"strmprivacy/example/1.3.0","nonce":{"int":1699175479},"timestamp":{"long":1652181101896},"keyLink":{"string":"5afc9c0a-de67-4e38-bd0a-66640f4349c7"},"billingId":{"string":"strmprodccdtest1908747604"},"consentLevels":[0,1,2,3]},"uniqueIdentifier":{"string":"ASotb1YzBuuBBM981rLIzQd/EZA7Em7dxyaBg7vE"},"consistentValue":"ASotb1bYf1G/2ye4h6ELE4hlkVGZGKSt3LQe0AnZU+k=","someSensitiveValue":{"string":"ASotb1bb51OCfM9HdCqcAjTLtXqK67EuBQicTpNzIvv/"},"notSensitiveValue":{"string":"not-sensitive-95"}}
+
+
+### Exporting to an s3 bucket
+
+Create the s3 bucket and the associated `sink` (soon to be renamed to `data-connector`). See [docs][sinks] for details
```
Comment (jankeesvanandel) [inline docs/quickstart/ccd.md:193]: Isn't data-connector the name already?

File: docs/quickstart/ccd.md [hunk 21]
```diff
@@ -0,0 +1,246 @@
...
+
+    strm create batch-exporter test-2 --sink s3 --path-prefix ccd-prod-ovh
+
+You should see a batch-exporter come to live:
+
+    kubectl get deployments.apps  -l app=batch-exporter
+    NAME                                                  READY   UP-TO-DATE   AVAILABLE   AGE
+    batch-exporter-c726bbff-2b68-4438-ac0d-ee877aa4dabf   1/1     1            1           2m49s
+
+If you're running data, you should see files in your s3 bucket quickly
```
Comment (jankeesvanandel) [inline docs/quickstart/ccd.md:206]: Is "running data" a common way of saying that it works?

**Review by jankeesvanandel — CHANGES_REQUESTED** [review]
